pkg/gadgets/trace/tcpconnlat/tracer/bpf/tcpconnlat.bpf.c

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2020 Wenbo Zhang
+#include <vmlinux/vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include "tcpconnlat.h"
+
+#define AF_INET    2
+#define AF_INET6   10
+
+// we need this to make sure the compiler doesn't remove our struct
+const struct event *unusedevent __attribute__((unused));
+
+const volatile __u64 targ_min_ns = 0;
+const volatile pid_t targ_tgid = 0;
+const volatile bool filter_by_mnt_ns = false;
+
+struct piddata {
+	char comm[TASK_COMM_LEN];
+	u64 ts;
+	u32 tgid;
+	u32 pid;
+	u64 mntns_id;
+};
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, 4096);
+	__type(key, struct sock *);
+	__type(value, struct piddata);
+} start SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+	__uint(key_size, sizeof(u32));
+	__uint(value_size, sizeof(u32));
+} events SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, 1024);
+	__uint(key_size, sizeof(u64));
+	__uint(value_size, sizeof(u32));
+} mount_ns_filter SEC(".maps");
+
+static int trace_connect(struct sock *sk)
+{
+	u64 pid_tgid = bpf_get_current_pid_tgid();
+	u32 tgid = pid_tgid >> 32;
+	u32 pid = (pid_t)pid_tgid;
```

Comment:
```suggestion
	u32 pid = (u32)pid_tgid;
```
(other gadgets do a simple cast)